### DISCUSSION TITLE
[Feature Request] InlineEdit a short list of related entities

### DISCUSSION THREAD

**Discussion by DinoTD** [Laravel-Backpack/community-forum#314] [General]
# Feature Request

### What's the feature you think Backpack should have?
The same as InlineCreatte but also with InlineEdit.

We have two tables here: orders and invoices. One order has zero or a short list of invoices. An invoice belongs to an order.
In an order create/update page I have an select(from a short list of invoices) with InlineCreate and a modal.
I want to transform it in a custom field as a list with InlineCreate and InlineEdit. This looks similar with a table field.
In this create/update page I want to see an "+ Add invoice" and also a short list of existing invoices with an "Edit" button for every invoice and the possibility to edit each invoice in a modal as I do now with InlineCreate.
How hard is to separate the logic from this select and InlineCreate.
What is the best way to do that? 

### Have you already implemented a prototype solution, for your own project?
For now I'm trying to implement this as a custom field for create and update.

### Do you see this as a core feature or an add-on?

It can be a core feature next to InlineCreate.

**Comment by welcome** [comment]
Hello there! Thanks for opening your first issue on this repo!

Just a heads-up: **Here at Backpack we use Github Issues only for tracking bugs**. Talk about new features is also acceptable. This helps _a lot_ in keeping our focus on improving Backpack. **If you issue is not a bug/feature, please help us out by closing the issue yourself and posting in the appropriate medium (see below)**. If you're not sure where it fits, it's ok, a community member will probably reply to help you with that.

Backpack communication channels:
 - Bug Reports, Feature Requests - Github Issues (here);
 - Quick help (_How do I do X_) - [Gitter Chatroom](https://gitter.im/BackpackForLaravel/Lobby);
 - Long questions (_I have done X and Y and it won't do Z wtf_) - [Stackoverflow](https://stackoverflow.com/questions/tagged/backpack-for-laravel), using the ```backpack-for-laravel``` tag;
 - Showing off something you've made, asking for opinion on Backpack/Laravel matters - [Reddit](https://www.reddit.com/r/BackpackForLaravel/);

Please keep in mind Backpack offers no official / paid support. Whatever help you receive here, on Gitter, Slack or Stackoverflow is thanks to our awesome _awesome_ community members, who give up some of their time to help their peers. If you want to join our community, just start pitching in. We take pride in being a welcoming bunch.

Thank you!

--
Justin Case
The Backpack Robot

**Comment by tabacitu** [comment]
@DinoTD I'll let @pxpm answer this since InlineCreate is his baby, he knows best how difficult it would be to create an InlineEdit. I'll move this to Discussions though, since it's a Feature Request. It's multi-thread so it's easier to come up with solutions.

**Reply by DinoTD** [reply]
I did a basic version but it is not working properly. 
The hard part is to separate the js logic from select field and the logic from the modal.

**Reply by tabacitu** [reply]
Indeed this is a disadvantage of the way we've _already_ implemented InlineCreate, and of the technology stack (the jQuery part of it, at least) - that complicated features like this one are pretty much bound to their interface components. It sucks, but... it's the way it is.

There are ways to reduce this, though. For example, instead of doing PHP things in the Model, you can do them inside the CrudPanel object; the CrudPanel object is macroable, so [you can add functions to it](https://backpackforlaravel.com/docs/4.1/crud-operations#adding-methods-to-the-crudpanel-object) by doing something like:

```php
       $this->crud->macro('getColumnsInTheFormatIWant', function() {
            $columns = $this->columns();
            // ... do something to $columns;
            return $columns;
        });
```

(though I don't quite understand what new model operations should happen... as far as I understand just the regular Edit should happen, right... just through a new route - maybe)

If you can share your progress so far as a PR, I'm sure @pxpm can take a look and give pointers if you're far along.

**Comment by jcastroa87** [comment]
Hi @DinoTD,

Thanks for suggesting this! We agree that having an InlineEdit option, similar to InlineCreate, would be a powerful addition for use cases like yours with orders and invoices.

The good news is that we plan to work on this feature during the development of Backpack v7. While we can’t commit to an exact release date just yet—there’s quite a bit of work ahead—it’s definitely on our backlog. We’re excited to bring more flexibility to Backpack’s inline functionalities and enhance user experience in these scenarios.

In the meantime, if you’re exploring custom implementations, feel free to share your progress with the community. We’re always interested to see how creative solutions can help shape Backpack’s future features.

Thanks again for the valuable feedback!

Cheers.
